miner/src/pool/scoring.rs

```diff
@@ -92,33 +92,69 @@ impl<P> txpool::Scoring<P> for NonceAndGasPrice where P: ScoredTransaction + txp
 	}
 
 	fn update_scores(&self, txs: &[txpool::Transaction<P>], scores: &mut [U256], change: scoring::Change) {
-		use self::scoring::Change;
-
-		match change {
-			Change::Culled(_) => {},
-			Change::RemovedAt(_) => {}
-			Change::InsertedAt(i) | Change::ReplacedAt(i) => {
-				assert!(i < txs.len());
-				assert!(i < scores.len());
-
-				scores[i] = *txs[i].transaction.gas_price();
-				let boost = match txs[i].priority() {
-					super::Priority::Local => 15,
-					super::Priority::Retracted => 10,
-					super::Priority::Regular => 0,
-				};
-				scores[i] = scores[i] << boost;
+
+		let boost = |i: usize| {
+			match txs[i].priority() {
+				super::Priority::Local => 15,
+				super::Priority::Retracted => 10,
+				super::Priority::Regular => 0,
+			}
+		};
+
+		let event = |scores: &mut [U256]| {
+			for (score, tx) in scores.iter_mut().zip(txs) {
+				if !tx.priority().is_local() {
+					*score = *score >> 3;
+				}
+			}
+		};
+
+		match self.0 {
+			PrioritizationStrategy::GasPriceOnly => {
+				match change {
+					Change::Culled(_) => {},
+					Change::RemovedAt(_) => {}
+					Change::InsertedAt(i) | Change::ReplacedAt(i) => {
+						// calculate base score + boost
+						assert!(i < txs.len());
+						assert!(i < scores.len());
+						scores[i] = *txs[i].transaction.gas_price() << boost(i);
+					},
+					// We are only sending an event in case of penalization.
+					// So just lower the priority of all non-local transactions.
+					Change::Event(_) => event(scores),
+				}
 			},
-			// We are only sending an event in case of penalization.
-			// So just lower the priority of all non-local transactions.
-			Change::Event(_) => {
-				for (score, tx) in scores.iter_mut().zip(txs) {
-					// Never penalize local transactions.
-					if !tx.priority().is_local() {
-						*score = *score >> 3;
+			PrioritizationStrategy::Consecutive => {
+				let is_consecutive = |i: usize| {
+					if (txs[i].nonce() + U256::one()) == txs[i+1].nonce() {
+						U256::one()
+					} else {
+						U256::zero()
 					}
+				};
+				match change {
+					Change::Culled(_) => {},
+					Change::RemovedAt(_) => {}
+					Change::InsertedAt(i) | Change::ReplacedAt(i) => {
+						assert!(i < txs.len());
+						assert!(i < scores.len());
+						let last_index = txs.len() - 1;
+						scores[last_index] = *txs[last_index].transaction.gas_price();
```

Comment:
I think you should include `boost` here

Reply:
Actually, currently `boost` seems to have exponential effect, can you instead apply it either:
1. Doing an additional loop after you calculate base scores (that will boost the scores, but boosting `idx+1` won't affect `idx`)
2. Instead of using `gas_price` use `gas_price` with boost applied.

I'm more inclined to option (2). Would be good to plot some graphs how they differ.

Reply:
This is something I considered doing before but wasn't sure if it was necessary, plotting all the different algorithms now! :+1:

Reply:
Seems that the 3rd graph has a different scale.

Note that we onyl care about transactions above 21k (base gas), would be good to plot starting from there.
Do you mind plotting all options on the same chart and add a line for regular scoring as well (i.e. the one with gas price only?)

Reply:
Merged all the graphs into two, one where Gas price was kept constant and another where Gas was kept at a constant. 


Legend for all the graphs:
Algorithm from PR:  Black,
Option 1: Blue
Option 2: Red
GasPriceOnly Scoring:  Green

## Constant Gas Price. Gas price kept at 10,000

![Constant Gas Price](https://raw.githubusercontent.com/insipx/eth-graphs/master/Gas.svg?sanitize=true)

#### Same graph, but with an extended X-Axis
![Constant Gas Price](https://raw.githubusercontent.com/insipx/eth-graphs/master/Gas_longx.svg?sanitize=true)

## Constant Gas. Gas kept at 21,000
![Constant Gas](https://raw.githubusercontent.com/insipx/eth-graphs/master/Gas_Price.svg?sanitize=true)